### PR TITLE
Support running Fred2 on Linux using Wine

### PR DESCRIPTION
**Pull request by Shivansps**
Support downloading and running Fred2 on Linux using Wine

-No default install, if someone wants to install Fred2 on Linux needs to "Modify" the FSO build and install the Windows Package.
-Arch limitation, Knet on linux will only consider executables of the same CPU arch as valid, that means, in X64 Linux, you cant download or run 32 bit Fred2.

image

### CODE REVIEW

File: Knossos.NET/Classes/Wine.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        {
+            try
+            {
+                var wineArch = GetWineArch(fsoArch);
+                if (wineArch == null)
+                {
+                    return new WineResult(false, "Unsupported WINEARCH: " + fsoArch.ToString());
+                }
+
+                var prefixResult = await SetWinePrefixFred2(wineArch);
```
Comment (BMagnu) [inline Knossos.NET/Classes/Wine.cs:43]: Should this really be called every time you launch FRED?
Could this not early exit if the requested WINEPREFIX already exists or something?
I would imagine that the wineboot is pretty much a no-op if the prefix exists, so it wouldn't be too much of an issue to keep it as is though.

**Review by BMagnu — COMMENTED** [review]
Looks mostly good. One question though.

File: Knossos.NET/Classes/Wine.cs (hunk 1 above)
Comment (Shivansps) [reply Knossos.NET/Classes/Wine.cs:43]: I dont know how to detect if it completed successfully previusly what if it did error out? Running it again does nothing so i made it run every time.

Comment (BMagnu) [reply Knossos.NET/Classes/Wine.cs:43]: I see, makes sense. 
(purely in theory, this wineboot isn't necessary, just running wine on a new prefix works fine, but that doesn't set the DLL-override and will make the mono prompt show)

**Review by BMagnu — APPROVED** [review]

**Comment by JohnAFernandez** [comment]
I honestly don't know what I'm looking at.  Based on BMagnu's approval and testing already done, I think we should merge this PR and release to wider testing.

**Comment by topas-rec** [comment]
I wanna leave a big thanks to you all for doing this. Having a separate wine prefix in `.config/KnossosNET/wine` is also fine in comparison to my suggestion to have a configurable wine prefix path.
Having the command line switches added for the selected mod in Knossos to the FRED2 launch is awesome. This way the used folders of the mod are found correctly in FRED2. I  thought I have to manually do something here, which I now don't have to.

> I honestly don't know what I'm looking at. Based on BMagnu's approval and testing already done, I think we should merge this PR and release to wider testing.

Works here!

An enhancement could be to guide Linux users to download the Windows build to get FRED2. This is not intuitive - but I already know it so no problem for me.   
Currently Linux users are just prompted with something like : THe executable for Fred2 wasn't found. I suggest to add something like "Please download a windows engine build".   

But my intention to come here was to say thanks.

**Pull request merged**